src/dotnet/commands/dotnet-add/dotnet-add-proj/Program.cs

```diff
@@ -104,9 +105,93 @@ private void AddProject(SlnFile slnFile, string projectPath)
                 };
 
                 slnFile.Projects.Add(slnProject);
+
+                EnsureBuildConfigurationsAreValidAndComplete(slnFile);
+
                 Reporter.Output.WriteLine(
                     string.Format(CommonLocalizableStrings.ProjectAddedToTheSolution, projectPath));
             }
         }
+
+        private void EnsureBuildConfigurationsAreValidAndComplete(SlnFile slnFile)
+        {
+            HashSet<string> _buildConfigurations = new HashSet<string>()
+            {
+                "Debug",
+                "Release",
+            };
+
+            HashSet<string> _buildPlatforms = new HashSet<string>()
+            {
+                "Any CPU",
+                "x64",
+                "x86",
+            };
+
+            var invalidConfigs = new List<string>();
+            foreach (var configPlatformCombo in slnFile.SolutionConfigurationsSection.Keys)
+            {
+                var configPlatformComponents = configPlatformCombo.Split('|');
+                if (configPlatformComponents.Length != 2)
+                {
+                    invalidConfigs.Add(configPlatformCombo);
+                }
+                else
+                {
+                    var config = configPlatformComponents[0].Trim();
+                    if (!_buildConfigurations.Contains(config))
+                    {
+                        _buildConfigurations.Add(config);
+                    }
+
+                    var platform = configPlatformComponents[1].Trim();
+                    if (!_buildPlatforms.Contains(platform))
+                    {
+                        _buildPlatforms.Add(platform);
+                    }
+                }
+            }
+
+            foreach (var invalidConfig in invalidConfigs)
+            {
+                slnFile.SolutionConfigurationsSection.Remove(invalidConfig);
```

Comment:
so if I create my own configuration MyDebugSuperBuild it will get removed when adding something to sln? is that expected